### ISSUE TITLE
Server: Severe scalability problem

### ISSUE THREAD

**Issue by norgoe** [Fmstrat/ownnote#239]
Hi,
I am running owncloud 8.1.3 with the current ownnote plugin on vm using 3 of the 4 cores of my xeon and 3GB of RAM.

I imported about 1800 Notes from evernote to ownnote.
And even days after the conversion is finished ownnote still is unusably slow.
Opening the notes page takes about 1 minute before notes are displayed.
If I delete a note (which by the ways should be possible if the note is open, too) it takes about 60seconds with almost no cpu usage until the UI is updated. Same happens on chaning a note.

As the import process lost most of my categorizations I would have to sort my notes again, which would take roughly 1800x2 minutes ....

**Comment by Fmstrat** [comment]
Hmm.. I've done some 1K notes tests with no issues on my system with lower specs than that. When doing any activities, what process is pegging the CPU? Also, do you have any notes that have really large amounts of content in them (such as lots of images)?

**Comment by norgoe** [comment]
All the notes were imported from an evernote export.
The Notes durectory is currently hosting 1060 Files and 1059 directories with surplus content not deeleted during the import.
Totalling in 1.8G of storage.
Many of the notes are clips of whole webpages, so I guess many of them include multiple images.

Two of the thre cores assigned to owncloud are used between 70-80% by mysqld when I open the Notes view.

**Comment by Fmstrat** [comment]
Is the import folder still there? If so, part of the slowness could be that it's scraping that and comparing to the DB every time you load the app. Could you try moving that out and see what happens? Also, if you are set to DB and Filesystem, could you try setting to DB only temporarily and let me know if the load/time is still the same or better? Thanks.

**Comment by norgoe** [comment]
Yes, all the data in the notes dir is still there. Maybe I misunderstood the import instructions, but as far as I remember it included something like "you can remove ... after all html files are renamed to htm".

Switching to db only did not change a thing regarding loading time.
Afterwards I renamed the Notes Folder to check if this helps.
After more than 10 minutes the notes screen loads again and miraculously the Notes Folder exists again with only htm files in it - no idea why as I switched to db only.

At least now the notes tab loads within 2 seconds.
But with 1000+ notes in it that formerly where sorted by folders in evernote and now are totally unsorted it is really hard to find anything again. And it is especially cumbersome to delete a note as there 
a) is no preview
b) no delete button when the note is opened
